### DISCUSSION TITLE
How to maintain order of execution of listeners in case of multiple listeners

### DISCUSSION THREAD

**Discussion by mishrasangeeta87** [apache/airflow#31660] [Q&A] [answered]
Hello,

I have installed openlineage plugin which uses listener feature [OpenLineage Integration Airflow](https://github.com/OpenLineage/OpenLineage/tree/main/integration/airflow) . I also want to add my own listener. So in this case how do we control the order of execution. Say if I want openlineage listener code to execute first and then want my listener plugin to execute during the events, is it possible?

**Comment by potiuk** [comment]
@mobuchowski ?

**Comment by mobuchowski** [comment, marked as answer]
@mishrasangeeta87 I know this: https://pluggy.readthedocs.io/en/latest/#call-time-order

And https://github.com/pytest-dev/pluggy/issues/250

So, I haven't tried it, but marking your `hookimpl` with

```
    @hookimpl(trylast=True)
```
should work in this case. 

In general, we don't have ordering mechanism for plugins other than this in Airflow.
If we needed it, it could be done - by controlling order in which plugins are registered, instead of doing it eagerly.

**Reply by mishrasangeeta87** [reply]
@mobuchowski Thanks a lot for the pointers.  This helps.
